plasTeX/Imagers/pdf2svg.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+
+import subprocess
+import re
+import shutil
+from plasTeX.Imagers import VectorImager
+
+width_re = re.compile('width="([0-9\\.]*)')
+height_re = re.compile('height="([0-9\\.]*)')
+class PDFSVG(VectorImager):
+    """ Imager that uses dvisvgm """
+    fileExtension = '.svg'
+    verifications = ['pdflatex --help', 'which pdf2svg', 'pdfcrop --help']
+    compiler = 'pdflatex'
+
+    def executeConverter(self, output):
+        open('images-uncropped.pdf', 'wb').write(output)
+        subprocess.call(["pdfcrop", "images-uncropped.pdf", "images.pdf"], stdout=subprocess.DEVNULL)
+
+        rc = 0
+        page = 1
+        while 1:
+            filename = 'img%d.svg' % page
+            rc = subprocess.call(['pdf2svg', 'images.pdf', filename, str(page)], stdout=subprocess.DEVNULL)
+            if rc:
+                break
+    
+            scale = self.config["images"]["scale-factor"]
+            if scale == 1:
+                continue
+
+            scaled = False
+            with open("scaled-%s" % filename, "w") as ofile:
+                for line in open(filename, "r"):
+                    if not scaled and "<svg" in line:
```

Comment:
All this SVG parsing seems brittle (see also the first answer on [this page](https://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags)). In the current TikZ handling we use BeautifulSoup. Maybe this is a bit extreme, but there is xml parsing in the python standard lib.

Reply:
I agree. My original rationale was that we only had to parse very specific output, namely what pdf2svg gives us, but I suppose this doesn't excuse bad code. I'll fix this.